src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/TimestampRequest.cs

```diff
@@ -11,18 +12,31 @@ namespace NuGet.Packaging.Signing
     public class TimestampRequest
     {
         /// <summary>
-        /// Signature that needs to be timestamped.
+        /// Signing Specification for this timestamp request.
+        /// </summary>
+        public SigningSpecifications SigningSpec { get; }
+
+        /// <summary>
+        /// Hash of Signature that needs to be timestamped.
         /// </summary>
-        public byte[] Signature { get; set; }
+        public byte[] SignatureMessageHash { get; }
 
         /// <summary>
         /// Hash algorithm to be used for timestamping.
         /// </summary>
-        public HashAlgorithmName TimestampHashAlgorithm { get; set; }
+        public HashAlgorithmName TimestampHashAlgorithm { get; }
```

Comment:
Can you change the above two properties as follows?  This aligns our terminology with [RFC 3161](v) and more accurately captures what these properties represent:

```
        /// <summary>
        /// Gets the hashed message to be timestamped.
        /// </summary>
        public byte[] HashedMessage { get; }

        /// <summary>
        /// Gets the hash algorithm used to generate <see cref="HashedMessage" />.
        /// </summary>
        public HashAlgorithmName HashAlgorithm { get; }
```

Update the constructor too.